### PR TITLE
Feature: Move open with to main menu when right clicking recent files

### LINKED ISSUE #11298: Feature: Move "open with" to main menu when right clicking recent files
### What feature or improvement do you think would benefit Files?

The recent files widget has an option to "open with" but it's in the overflow menu. Moving it to the main menu would improve the user experience and consistency between the layouts.

### Requirements

- This proposal will display "open with" in the main menu when right clicking in the recent files widget

### Files Version

v2.4.34

### Windows Version

Windows 11

### Comments

_No response_

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #11298

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Comment by yaira2** [comment]
Is this working for you? I don't see the option and it's crashing as well.

**Comment by hecksmosis** [comment]
This works fine for me

**Comment by hecksmosis** [comment]
One of the few times CI doesn't spontaneously combust

**Comment by yaira2** [comment]
It's still not working for me, maybe someone else can test

**Comment by hecksmosis** [comment]
I cannot reproduce this, don't know why this is happening

**Comment by hecksmosis** [comment]
Does this happen in main?

**Comment by yaira2** [comment]
`main` is fine

**Comment by yaira2** [comment]
@hishitetsu does this work for you?

**Comment by yaira2** [comment]
@hecksmosis in the meantime, can you share a screen recording of this in action?

**Comment by hishitetsu** [comment]
Sorry, I'm away from home and can't debug right now, but `showOpenWithMenu: true` was not specified in the other widgets and was causing the shell extension not to load, so I removed this option.

**Comment by hecksmosis** [comment]
Oh, that's why then

**Comment by hecksmosis** [comment]
@hishitetsu I looked into this and it seems to work but only if you don't await it

**Comment by hecksmosis** [comment]
I'll send a screen recording

**Comment by hecksmosis** [comment]
[f1fcd40d-7694-4ae8-b9c3-5314700cc655.webm](https://user-images.githubusercontent.com/55754091/219298289-5a824c84-5f1b-4d53-9ea1-3e1c2356fb9d.webm)

**Comment by hishitetsu** [comment]
I have identified the cause of the error.
Once #11325 is merged, the error should no longer occur.

**Comment by yaira2** [comment]
It's working now but it looks like it's missing the placeholder that's supposed to show when the sub menu is loading. Also, there seems to be an issue in menu flyouts where the colored icons don't work so we might have to use a regular icon for now.

**Comment by hecksmosis** [comment]
> It's working now but it looks like it's missing the placeholder that's supposed to show when the sub menu is loading. Also, there seems to be an issue in menu flyouts where the colored icons don't work so we might have to use a regular icon for now.

@yair100 Do you mean a placeholder for the open with menu so it doesn't change size?

**Comment by yaira2** [comment]
Yes

**Comment by hecksmosis** [comment]
Ok, on it

**Comment by hecksmosis** [comment]
What icon should we use?

**Comment by yaira2** [comment]
`\uE17D`

**Comment by hecksmosis** [comment]
Does this have a color part or should I use \uF047 and \uF048? (These are the ones used in BaseLayout)

**Comment by hecksmosis** [comment]
Done

File: src/Files.App/Helpers/ShellContextMenuHelper.cs [hunk 1]
```diff
@@ -255,7 +255,15 @@ public static async Task LoadShellMenuItems(string path, CommandBarFlyout itemCo
 					var openWithItem = shellMenuItems.Where(x => (x.Tag as Win32ContextMenuItem)?.CommandString == "openas").ToList().FirstOrDefault();
 					if (openWithItem is not null)
 					{
+						openWithItem.ColoredIcon = new ColoredIconModel()
+						{
+							BaseLayerGlyph = "\uF049",
+							OverlayLayerGlyph = "\uF04A",
+						};
```
Comment (yaira2) [inline src/Files.App/Helpers/ShellContextMenuHelper.cs:262]: Oh, color icon is working here?

**Review by yair100 — APPROVED** [review]
LGTM

File: src/Files.App/Helpers/ShellContextMenuHelper.cs (hunk 1 above)
Comment (hecksmosis) [reply src/Files.App/Helpers/ShellContextMenuHelper.cs:262]: Found out about this and decided to try it

Comment (hecksmosis) [reply src/Files.App/Helpers/ShellContextMenuHelper.cs:262]: It's added in the MenuFlyoutItem before it's converted into an AppBarButton

Comment (yaira2) [reply src/Files.App/Helpers/ShellContextMenuHelper.cs:262]: Can we do the same for the "new" flyout? We have a colored icon for the shortcut option but it wasn't working to use the icon there.

**Pull request merged**